### DISCUSSION TITLE
ruleset xml arguments

### DISCUSSION THREAD

**Discussion by Research-SE-Otago** [pmd/pmd#3551] [Q&A] [answered]
Will you be able to help me with defining the required arguments to new PMD version.  The links by default seems old and URL s not working. I am using the rules appears on this new version (PMD https://pmd.github.io/latest/index.html), seems it is PMD 6.39.0. Could you please help me with the arguments that I should pass to below?

xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">

Thanks very much !!

**Comment by adangel** [comment]
Which error do you get and which URL exactly is not working?

**Reply by Research-SE-Otago** [reply]
The URL: http://pmd.sourceforge.net/ruleset/2.0.0 . Should this be a working URL?

In brief , all I want to know is whether below schema and schema location refer to the updated PMD (i.e., 6.39.0) rule set.   
xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">

Sorry, I am new to this and have never used the tool.

**Comment by adangel** [comment, marked as answer]
No, "http://pmd.sourceforge.net/ruleset/2.0.0" is not a URL, it's just a URI. This is the namespace. The XML schema is located at  <http://pmd.sourceforge.net/ruleset_2_0_0.xsd> (which redirects to <https://pmd.sourceforge.io/ruleset_2_0_0.xsd>).

> updated PMD (i.e., 6.39.0) rule set

The above xml prolog only specifies the format (namespace, schema). There is no "updated PMD rule set".

You can start with the [quickstart ruleset](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/resources/rulesets/java/quickstart.xml) and modify this or create a new ruleset as described in [Making rulesets](https://pmd.github.io/latest/pmd_userdocs_making_rulesets.html).
It's your decision, which rules you enable. Available rules can be found here: https://pmd.github.io/latest/pmd_rules_java.html

**Reply by Research-SE-Otago** [reply]
Many thanks for the detailed explanation. Much appreciated.!!
